tests_new/integration_tests/client.py

```diff
@@ -28,15 +32,18 @@ def __init__(self, username, password):
         wait_random_min=1000,
         wait_random_max=3000,
     )
-    def query(self, query: str):
+    def query(self, query: dict):
         graphql_endpoint = os.path.join(os.environ['API_ENDPOINT'], 'graphql', 'api')
         headers = {'accesskeyid': 'none', 'SecretKey': 'none', 'Authorization': f'Bearer {self.access_token}'}
         r = requests.post(graphql_endpoint, json=query, headers=headers)
-        if errors := r.json().get('errors'):
-            raise GqlError(errors)
+        response = r.json()
+        if errors := response.get('errors'):
+            if any((response.get('data', {}) or {}).values()):  # check if there are data
```

Comment:
Why do we need to check for `.values()`? wouldn't it be enough with `if any((response.get('data', {}):`

Reply:
It won't be enough, a GQL response looks like this so we have to check if subkeys (query1,query2,...) are non empty
```json
{
"data" : {
  "query1": {"key1": "data..."}
  "query2": None
}
```